apps/redi-talent-pool/src/utils/location-details.ts

```diff
@@ -0,0 +1,14 @@
+export const locationDetails = (
+  company: string,
+  city: string,
+  country: string
+): string => {
+  if (company && city && country) return `${company} (${city}, ${country})`
+  if (company && (city || country))
+    return `${company} (${city ? city : country})`
+  if (city && country) return [city, country].join(', ')
+  if (company) return company
+  if (city) return city
+  if (country) return country
+  return null
+}
```

Comment:
This is a very good idea, to abstract this function and reuse it in two places. How about improving several things here?

1) We can be more specific with the name. `locationDetails` might not ring the bell for everyone that it receives some params and returns a string. (Of course, TypeScript solves this problem partially). I'd go with something like `getInstitutionAndLocationString`. Nothing is wrong with being too explicit. (Point 2 explains why I suggest using Institution in the name)

2) We're calling the first parameter `company` in this function but reusing this also for Education, where we pass `institutionName` instead. Since I don't have a better word, I feel like even `institution` is better to be used here, since companies are also institutions. But if you're good at naming and if you have a better word to use both for companies and educational institutions, feel free to use that. Also, updating the name I suggest in 1).

3) If you're gonna implement my two comments above (not rendering the paragraph, if the company or institution name is missing), we can also return null here when the first parameter is missing. So in this way, you can make this a tiny bit more readable:

```suggestion
): string => {
  if (!company) return null;
  
  if (city && country) return `${company} (${city}, ${country})`
  
  if (city || country) return `${company} (${city || country})`
  
  return company;
}
```

There are actually many more ways to achieve the same result. Let me know if you don't like this one so we can look at other options :v:

Reply:
Can we also find a better place for this piece of code to live? `utils` to me really means "miscellaneous utility functions that are generally used across the entire app". Since this function is so far only used by components in `apps/redi-talent-pool/src/components/organisms/jobseeker-profile-editables`, how about we put the file in the same folder?

In short; long live [colocation](https://kentcdodds.com/blog/colocation).

Reply:
On the same note; @helloanil, should we do a quick refactor of the confetti code to another folder? :D

Reply:
Hmmm, perhaps we should even turn the location code into a react component that returns a fragment? Perhaps by making it a React component we make it even more explicit that "yep, this is view-layer code".

Reply:
I implemented all the requests changes because I liked the ideas and approaches. I also added schemas to make a few inputs required so the record makes sense to a) be displayed at all and b) don't ruin the design once it's saved. 

Let me know what you guys think!

Reply:
@marcuszierke What do you think about following @ericbolikowski's last suggestion and turning `getInstitutionAndLocationString` function into a React Component. Something like `InstitutonAndLocation` might work. We can pass the same params (company, country, city or institution, country, city) to this component and it would return the whole `<p>` thing including the style.